canonical/lxd-ui
# Fixes inability to delete single profiles + minor UI change - [WD-10834]

## Done

- "Add Profile" button now on a new line whenever there are no profiles in an instance (UI change)
- Instances with single profiles can now delete the profile as long as an alternative disk device is specified, or the default disk device, overridden.
- The above applies to when instances are being created and when they are being edited.
- Changes have only been made to ProfileSelector.tsx.

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @mas-who or @edlerd for access.
    - With a local copy of this branch, run as described [here](../HACKING.md#setting-up-for-development).
2. Perform the following QA steps:
    - To test this fix, navigate to an existing instance -> "Edit Instance" -> View the ability to delete a single instance.
    - Attempting to save an instance with no profile and no alternative disk device specified should result in an error, and an inability to save the changes.
    - One can also create a new instance and view the same feature. Both the instance creation and instance editing pages have the same component.

## Screenshots

![image](https://github.com/canonical/lxd-ui/assets/58699506/302d078b-2fa4-4a6e-a1fb-50ca626d0d49)

![image](https://github.com/canonical/lxd-ui/assets/58699506/43bd8a28-9dde-48b9-a2ff-19270260052e)
